framework/src/variables/MooseVariableFE.C

```diff
@@ -863,7 +863,8 @@ typename MooseVariableFE<OutputType>::ValueType
 MooseVariableFE<OutputType>::evaluate(const NodeArg & node_arg, const StateArg & state) const
 {
   mooseAssert(node_arg.node, "Must have a node");
-  mooseAssert(this->hasBlocks(node_arg.subdomain_id),
+  mooseAssert(node_arg.subdomain_id == Moose::INVALID_BLOCK_ID ||
```

Comment:
You must be using `Moose::INVALID_BLOCK_ID` because of something I've done? The assert definitely looked better before. Changing this to `Moose::ANY_BLOCK_ID` would seem more correct, but again I'm guessing you've added this code because of something preexisting that I wrote. I wonder if we can do better

Reply:
in nodal kernels we tend not to know what subdomain id to use unfortunately. so I ve started writing nodal args with invalid block id, but then they cant be used for this

Reply:
If that's the case, I don't think functors should accept `INVALID_BLOCK_ID` as valid input. If an object using a functor isn't picky, I think putting `ANY_BLOCK_ID` as the subdomain ID is better

Reply:
interesting. That works for me!